niworkflows/anat/mni.py

```diff
@@ -164,35 +170,115 @@ def _config_ants(self, ants_settings):
+                out_files.append(out_file)
+
+                # Open generated svg file and fix id
+                with open(out_file, 'rb') as f:
+                    svg = f.read()
+
+                # Find and replace the figure_1 id.
+                xml_data = etree.fromstring(svg)
+                find_text = etree.ETXPath("//{%s}g[@id='figure_1']" % (SVGNS))
+                find_text(xml_data)[0].set('id', '%s-%s-%s' % (div_id, mode, uuid4()))
+
+                with open(out_file, 'wb') as f:
+                    f.write(etree.tostring(xml_data))
+            return out_files
+
+        def _compose_view(bg_svgs, fg_svgs, ref=0, out_file='report.svg'):
+            import svgutils.transform as svgt
+            import svgutils.compose as svgc
+
+            # Read all svg files and get roots
+            svgs = [svgt.fromfile(f) for f in bg_svgs + fg_svgs]
+            roots = [f.getroot() for f in svgs]
+            nsvgs = len(svgs) // 2
+            # Query the size of each
+            sizes = [(int(f.width[:-2]), int(f.height[:-2])) for f in svgs]
+
+            # Calculate the scale to fit all widths
+            scales = [1.0] * len(svgs)
+            if not all([width[0] == sizes[0][0] for width in sizes[1:]]):
+                ref_size = sizes[ref]
+                for i, els in enumerate(sizes):
+                    scales[i] = ref_size[0]/els[0]
+
+            newsizes = [tuple(size)
+                        for size in np.array(sizes) * np.array(scales)[..., np.newaxis]]
+
+            # Compose the views panel: total size is the width of
+            # any element (used the first here) and the sum of heights
+            totalsize = [newsizes[0][0], np.sum(newsizes[:3], axis=0)[1]]
+            fig = svgt.SVGFigure(totalsize[0], totalsize[1])
+
+            yoffset = 0
+            for i, r in enumerate(roots):
+                size = newsizes[i]
+                r.moveto(0, yoffset, scale=scales[i])
+                yoffset += size[1]
+                if i == (nsvgs - 1):
+                    yoffset = 0
+
+            # Group background and foreground panels in two groups
+            newroots = [
+                svgt.GroupElement(roots[:3], {'class': 'background-svg'}),
+                svgt.GroupElement(roots[3:], {'class': 'foreground-svg'})
+            ]
+            fig.append(newroots)
+            out_file = op.abspath(out_file)
+            fig.save(out_file)
+
+            # Add styles for the flicker animation
+            with open(out_file, 'rb') as f:
+                svg = f.read().split('\n')
+
+            # Replace header (get rid of height, width and viewBox) and add custom stylesheet
+            # svg[1] = ('<svg xmlns:xlink="http://www.w3.org/1999/xlink" '
+            #           'xmlns="http://www.w3.org/2000/svg" version="1.1" class="flickering" '
+            #           'viewBox="0 0 %0.2f %0.2f">' % tuple(totalsize))
+            svg.insert(2, """\
+  <style type="text/css">
```

Comment:
Ok, I'll add the unique string. But yeah, these svgs can be opened themselves and they correctly show the flickering trick since the CSS are embeded, I don't see why the styling should go aside.